src/main/java/labaSolvd/JacoLuna/Services/entityServices/ReviewsService.java

```diff
@@ -49,34 +67,196 @@ public void add() {
+                } catch (JAXBException e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+            case DATA_BASE -> {
+                try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+                    if (session.getMapper(ReviewMapper.class).deleteReview(id) > 0) {
+                        session.commit();
+                        result = true;
+                    }
+
+                } catch (Exception e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+        }
+        return result;
     }
 
     @Override
     public List<Review> getAll() {
-        if (source == SourceOptions.DATA_BASE)
-//            return reviewDAO.getList();
-            return null;
-        try {
-            return Marshaller.UnMarshallList(Reviews.class, XmlPaths.REVIEWS);
-        } catch (JAXBException e) {
-            Utils.CONSOLE_ERROR.error(e);
+        switch (source) {
+            case XML -> {
+                try {
+                    return Marshaller.UnMarshallList(Reviews.class, XmlPaths.REVIEWS);
+                } catch (JAXBException e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
+            case JSON -> {
+                return JsonParser.unparseToList(Review.class, JsonPaths.REVIEWS);
+            }
+            case DATA_BASE -> {
+                try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+                    return session.getMapper(ReviewMapper.class).getAllReviews();
+                } catch (Exception e) {
+                    Utils.CONSOLE_ERROR.error(e);
+                }
+            }
         }
         return null;
     }
 
     @Override
     public List<Review> search() {
-        return List.of();
+        List<Review> searchList = null;
+        try (SqlSession session = SessionFactoryBuilder.getSqlSessionFactory().openSession()) {
+            List<Field> attributes = Arrays.stream(Review.class.getDeclaredFields()).toList();
+            int attIndex = selectAtt(attributes);
+            Object value;
+
+            if (attributes.get(attIndex).getType().equals(String.class)) {
+                value = InputService.stringAns("Please enter the search value");
+                searchList = session.getMapper(ReviewMapper.class).searchByString(attributes.get(attIndex).getName(), (String) value);
+            } else {
+                Class<?> fieldType = attributes.get(attIndex).getType();
+                Utils.CONSOLE.info(fieldType.getName());
+                String prompt = "Please enter the search value";
+                value = InputService.setInput(prompt, Integer.class);
+
+                if (value != null) {
+                    searchList = session.getMapper(ReviewMapper.class).searchByNumber(attributes.get(attIndex).getName(), (Number) value);
```

Comment:
Following last comment:
```suggestion
            Class<?> fieldType = att.getType();
            if (fieldType.equals(String.class)) {
                value = InputService.stringAns("Please enter the search value");
                searchList = session.getMapper(ReviewMapper.class).searchByString(att.getName(), (String) value);
            } else {
                Utils.CONSOLE.info(fieldType.getName());
                String prompt = "Please enter the search value";
                value = InputService.setInput(prompt, Integer.class);

                if (value != null) {
                    searchList = session.getMapper(ReviewMapper.class).searchByNumber(att.getName(), (Number) value);
```